mypyc/primitives/int_ops.py

```diff
@@ -148,6 +148,14 @@ def int_unary_op(name: str, c_function_name: str) -> CFunctionDescription:
     c_function_name='CPyTagged_IsLt_',
     error_kind=ERR_NEVER)
 
+int_min = c_function_op(
+    name="builtins.min",
+    arg_types=[int_rprimitive, int_rprimitive],
+    return_type=int_rprimitive,
+    c_function_name='Py_MIN',
```

Comment:
`int_rprimitive` is actually `CPyTagged` in the generated C, i.e. a tagged pointer. `Py_MIN` only works with plain integers, so it only works for a subset of possible values. You should test also with a big integer, such as a value higher than 2**63.

There are two ways around this:

* Add a function `CPyTagged_Min(x, y)` to `mypyc/lib-rt/int_ops.c` that uses tagged integer operations. This is the easiest option but only works with integers. We'd need separate functions for all supported argument types.
* Inline the implementation of `min` by generating IR operations in `specialize.py`. You can use the `binary_op` method, for example, so that this works for arbitrary types.